lib/info.js

```diff
@@ -88,6 +87,48 @@ module.exports = function getInfo(link, opts, callback) {
 };
 
 
+function requestAndMergeMpd(id, config, callback) {
+  var infoUrl = getInfoUrl(id);
+  //the dashmpd for a vivo always returns a 403
+  //youtube-dl also ignores mpd download
+  if (config.args.ptk === 'vevo') {
+    callback();
+  } else {
+    if (!config.args.dashmpd) {
+      request(infoUrl, function(err, body) {
+        if (err) return callback(err);
+        var escaped = querystring.parse(body);
+        config.args.dashmpd = escaped.dashmpd;
+        callback();
+      });
+    } else {
+      callback();
+    }
+  }
+}
+
+/*
+youtube-dl 
+https://github.com/rg3/youtube-dl/blob/master/youtube_dl/extractor/youtube.py#L1127
+*/
+function getInfoUrl(id) {
```

Comment:
Since `requestAndMergeMpd()` is already a separated function, making another one for this that only gets called by it doesn't seem worth it.